src/Implementations/AzureTableStorage.cs

```diff
@@ -227,5 +230,64 @@ public async Task UpdateEntityAsync<T>(T oldEntity, T newEntity, string table) w
             TableOperation update = TableOperation.Replace(newEntity);
             await tableRef.ExecuteAsync(update, _options, null);
         }
+
+        /// <inheritdoc />
+        public async Task DeleteInBatchAsync(
+            string tableName,
+            DateTimeOffset cutOffDate,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            CloudTable table = await GetTableReferenceAsync(tableName);
+            TableQuery<DynamicTableEntity> query = new TableQuery<DynamicTableEntity>().Where(TableQuery.GenerateFilterConditionForDate("Timestamp", QueryComparisons.LessThan, cutOffDate));
+            IEnumerable<DynamicTableEntity> entities = await GetEntitiesAsync<DynamicTableEntity>(table, query, cancellationToken);
+            IEnumerable<TableBatchOperation> batches = GetBatchesForDelete(entities);
+            foreach (TableBatchOperation batch in batches)
+            {
+                await Task.WhenAll(table.ExecuteBatchAsync(batch));
+            }
+        }
+
+        private IEnumerable<TableBatchOperation> GetBatchesForDelete<T>(IEnumerable<T> tableEntities) where T : ITableEntity, new()
```

Comment:
This function looks to take items, group them by partition, and then put them in batches. Is there a reason why each partition couldn't be a single batch? Is there a need to make the batches smaller? Right now we put them in groups of 100...can you explain a bit why we do that? :)

Reply:
https://docs.microsoft.com/en-us/rest/api/storageservices/performing-entity-group-transactions